pkg/controller/common/certificates/http/reconcile.go

```diff
@@ -14,48 +14,51 @@ import (
 	"strings"
 	"time"
 
-	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
-	"github.com/elastic/cloud-on-k8s/pkg/controller/common/certificates"
-	"github.com/elastic/cloud-on-k8s/pkg/controller/common/driver"
-	"github.com/elastic/cloud-on-k8s/pkg/controller/common/name"
-	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
-	netutil "github.com/elastic/cloud-on-k8s/pkg/utils/net"
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/kubernetes/scheme"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/certificates/ca"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/certificates/certutils"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/name"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/watches"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	netutil "github.com/elastic/cloud-on-k8s/pkg/utils/net"
 )
 
 var (
-	log = logf.Log.WithName("http")
+	log = logf.Log.WithName("http_certs")
 )
 
 // ReconcileHTTPCertificates reconciles the internal resources for the HTTP certificate.
 func ReconcileHTTPCertificates(
-	driver driver.Interface,
+	k8sClient k8s.Client,
```

Comment:
I know it is not strictly a part of your PR, but, this function has just too many parameters. Maybe this refactoring is a good chance to change it.  It might be easier to read if a struct was used to pass the arguments instead. Of course, the downside is that it will make it harder for the compiler to catch problems if the function signature changes in the future. WDYT?

Reply:
yeah it's a trade-off I often struggle with.
On one hand the struct approach makes the function signature cleaner.
On the other hand, it's just as verbose anyway (actually more since we need to declare the struct), and it's fairly easy to forget initializing some of the fields. Whereas it's not possible through function arguments. I'll still try to see if I can do something about it.

Reply:
Wrapped them all in a struct. I think it looks cleaner 👍